src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.Editor.cs

```diff
@@ -0,0 +1,272 @@
+
+                // Get the original source document again, from the latest forked solution.
+                var sourceDocument = solutionWithNewDocument.GetDocument(SemanticDocument.Document.Id);
+
+                // update source document to add partial modifiers to type chain
+                // and/or remove type declaration from original source document.
+                var solutionWithBothDocumentsUpdated = await UpdateSourceDocumentAsync(
+                      sourceDocument, _state.TypeNode, _cancellationToken).ConfigureAwait(false);
+
+                return new CodeActionOperation[] { new ApplyChangesOperation(solutionWithBothDocumentsUpdated) };
+            }
+
+            private async Task<IEnumerable<CodeActionOperation>> RenameTypeToMatchFileAsync(Solution solution)
+            {
+                var symbol = _state.SemanticDocument.SemanticModel.GetDeclaredSymbol(_state.TypeNode, _cancellationToken) as INamedTypeSymbol;
+                var newSolution = await Renamer.RenameSymbolAsync(solution, symbol, _state.DocumentName, SemanticDocument.Document.Options, _cancellationToken).ConfigureAwait(false);
+                return new CodeActionOperation[] { new ApplyChangesOperation(newSolution) };
+            }
+
+            private IEnumerable<CodeActionOperation> RenameFileToMatchTypeName(Solution solution)
+            {
+                var text = SemanticDocument.Text;
+                var oldDocumentId = SemanticDocument.Document.Id;
+                var newDocumentId = DocumentId.CreateNewId(SemanticDocument.Document.Project.Id, _state.TargetFileNameCandidate);
+
+                var newSolution = solution.RemoveDocument(oldDocumentId);
+                newSolution = newSolution.AddDocument(newDocumentId, _state.TargetFileNameCandidate, text);
+
+                return new CodeActionOperation[]
+                {
+                    new ApplyChangesOperation(newSolution),
+                    new OpenDocumentOperation(newDocumentId)
+                };
+            }
+
+            private async Task<Solution> AddNewDocumentWithTypeDeclarationAsync(
+                SemanticDocument sourceDocument,
+                string newDocumentName,
+                DocumentId newDocumentId,
+                TTypeDeclarationSyntax typeNode,
+                CancellationToken cancellationToken)
+            {
+                var root = sourceDocument.Root;
+                var projectToBeUpdated = sourceDocument.Document.Project;
+                var documentEditor = await DocumentEditor.CreateAsync(sourceDocument.Document, cancellationToken).ConfigureAwait(false);
+
+                AddPartialModifiersToTypeChain(documentEditor, typeNode);
+
+                var membersToRemove = GetMembersToRemove(root, typeNode, removeDescendentsOfSelf: _makeTypePartial);
+                foreach (var member in membersToRemove)
```

Comment:
Think I might have cleared it by answering how we move types in a response to an earlier question, but here is a trivial example (there are more complex ones too) :
Move class2, which means everything else is to be removed in the new document after forking it from here.

```C#
namespace N1
{
    class Class1  {   }
   class Class2  {   }
   class Class3  {   }
}
namespace N2 { }

```